spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/DefaultChannelFactory.java

```diff
@@ -30,21 +29,33 @@
  */
 public class DefaultChannelFactory implements ChannelFactory {
 
-	@Autowired
-	MessageConverterConfigurer messageConverterConfigurer;
+	private final MessageConverterConfigurer messageConverterConfigurer;
+
+	public DefaultChannelFactory(MessageConverterConfigurer messageConverterConfigurer) {
+		this.messageConverterConfigurer = messageConverterConfigurer;
+	}
 
 	@Override
-	public MessageChannel createChannel(String name, Class<?> inputChannelType) throws Exception {
-		MessageChannel messageChannel = createMessageChannel(inputChannelType);
-		messageConverterConfigurer.configureMessageConverters(messageChannel, name);
-		return messageChannel;
+	public PollableChannel createPollableBindableChannel(String name) throws Exception {
+		PollableChannel pollableChannel = new QueueChannel();
+		messageConverterConfigurer.configureMessageConverters(pollableChannel, name);
+		return pollableChannel;
 	}
 
-	private MessageChannel createMessageChannel(Class<?> messageChannelType) {
-		return isPollable(messageChannelType) ? new QueueChannel() : new DirectChannel();
+	@Override
+	public SubscribableChannel createSubscribableBindableChannel(String name) throws Exception {
+		SubscribableChannel subscribableChannel = new DirectChannel();
+		messageConverterConfigurer.configureMessageConverters(subscribableChannel, name);
+		return subscribableChannel;
 	}
 
-	private boolean isPollable(Class<?> channelType) {
-		return PollableChannel.class.equals(channelType);
+	@Override
+	public SubscribableChannel createSubscribableSharedChannel() throws Exception {
```

Comment:
Maybe we can simplify this a little bit by elliminating the 'sharedChannel' methods and having just `createPollableChannel` and `createSubscribableChannel`. At this particular point, we don't configure them, and we can isolate bridging as a concern between the Aggregator and the BindableProxyFactory. Maybe we can rename the interface as `BindableChannelFactory` so it's clear it's used only for that. I believe we need to revisit configuration on aggregation anyway.